File: 20210829/26. Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array_kimhyeri.swift
//
//  Remove Duplicates from Sorted Array_kimhyeri.swift
//  Leetcode
//
//  Created by ELLIE on 2021/08/29.
//

import Foundation

class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        if nums.count <= 1 {
            return nums.count
        }
        
        var count = nums.count
        var left = 0
        var right = 1
        
        while right < count {
            if nums[left] == nums[right] {
                nums.remove(at: right)
                count = count - 1
            } else {
                left = left + 1
                right = right + 1
            }
        }
        
        return nums.count
    }
}

File: 20210829/26. Remove Duplicates from Sorted Array/Remove Duplicates from Sorted Array_kimhyeri.c
int removeDuplicates(int* nums, int numsSize){
    if(numsSize <= 1) {
        return numsSize;
    }
    
    int i = 1;
    int k = 1;
    
    for(i; i < numsSize; i++) {
        if(nums[i] != nums[i - 1]) {
            nums[k] = nums[i];
            k++;
        }
    }
    
    return k;
}
